goalie/mesh_seq.py

```diff
@@ -232,33 +317,46 @@ def _function_spaces_consistent(self) -> bool:
             )
         return consistent
 
-    def update_function_spaces(self) -> list:
+    def _update_function_spaces(self):
         """
-        Update the function space dictionary associated with the :class:`MeshSeq`.
+        Update the function space dictionary associated with the mesh sequence.
         """
-        if self._fs is None or not self._function_spaces_consistent:
-            self._fs = [self.get_function_spaces(mesh) for mesh in self.meshes]
+        if self._fs is None or not self._function_spaces_consistent():
             self._fs = AttrDict(
                 {
-                    field: [self._fs[i][field] for i in range(len(self))]
+                    field: [self.get_function_spaces(mesh)[field] for mesh in self]
                     for field in self.fields
                 }
             )
         assert (
-            self._function_spaces_consistent
+            self._function_spaces_consistent()
         ), "Meshes and function spaces are inconsistent"
-        return self._fs
 
     @property
     def function_spaces(self):
-        return self.update_function_spaces()
+        """
+        Get the function spaces associated with the mesh sequence.
+
+        :returns: a dictionary whose keys are field names and whose values are the
+            corresponding function spaces
+        :rtype: :class:`~.AttrDict` with :class:`str` keys and
+            :class:`firedrake.functionspaceimpl.FunctionSpace` values
+        """
+        self._update_function_spaces()
+        return self._fs
 
     @property
-    def initial_condition(self) -> AttrDict:
+    def initial_condition(self):
+        """
+        Get the initial conditions associated with the first subinterval.
+
+        :returns: a dictionary whose keys are field names and whose values are the
+            corresponding initial conditions applied on the first subinterval
+        :rtype: :class:`~.AttrDict` with :class:`str` keys and
+            :class:`firedrake.function.Function` values
+        """
         ic = OrderedDict(self.get_initial_condition())
```

Comment:
elsewhere you have changed `ic` to be `initial_condition_map`. Would it help consistency to do something similar here?

Reply:
Fixed in be9ed9e.